crates/pallet-humanode-session/src/lib.rs

```diff
@@ -72,6 +74,67 @@ pub mod pallet {
     #[pallet::storage]
     pub type CurrentSessionIndex<T: Config> = StorageValue<_, SessionIndex, OptionQuery>;
 
+    /// A list of all banned accounts that can't be validators in the network.
+    #[pallet::storage]
+    pub type BannedAccounts<T: Config> =
+        StorageValue<_, BoundedVec<T::AccountId, T::MaxBannedAccounts>, ValueQuery>;
+
+    /// Possible errors.
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Attempt to disable bootnode to be a validator.
+        AttemptToDisableBootnode,
+        /// Failed to convert `T::AccountId` to `T::ValidatorId`.
+        AccountIdToValidatorId,
+        /// The provided account could not be found in current validators list or it was already
+        /// disabled.
+        AccountIsNotValidator,
+        /// The account is already banned.
+        AccountIsAlreadyBanned,
+        /// The BannedAccounts storage has reached the limit as BoundedVec.
+        TooManyBannedAccounts,
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// Kick validator based on provided account id.
+        #[pallet::call_index(0)]
+        #[pallet::weight(0)]
+        pub fn kick(origin: OriginFor<T>, account_id: T::AccountId) -> DispatchResult {
+            ensure_root(origin)?;
+
+            Self::disable(account_id)?;
+
+            Ok(())
+        }
+
+        /// Ban account.
+        #[pallet::call_index(1)]
+        #[pallet::weight(0)]
+        pub fn ban(origin: OriginFor<T>, account_id: T::AccountId) -> DispatchResult {
+            ensure_root(origin)?;
+
+            ensure!(
+                !Self::is_banned(&account_id),
+                Error::<T>::AccountIsAlreadyBanned
+            );
+
+            with_storage_layer(move || {
+                Self::disable(account_id.clone())?;
+
+                <BannedAccounts<T>>::try_mutate::<_, DispatchError, _>(move |banned_accounts| {
```

Comment:
Why not try_push directly?

Reply:
Thanks, good suggestions - https://github.com/humanode-network/humanode/pull/1249/commits/c85780364ab73b6932cb581d829192e0fb5e7ebc.